### PR TITLE
community[patch]: fix public interface for embeddings module

### PR DESCRIPTION
**Pull request by ringohoffman**
## Description

The existing public interface for `langchain_community.emeddings` is broken. In this file, `__all__` is statically defined, but is subsequently overwritten with a dynamic expression, which type checkers like pyright do not support. pyright actually gives the following diagnostic on the line I am requesting we remove:

[reportUnsupportedDunderAll](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnsupportedDunderAll):

```
Operation on "__all__" is not supported, so exported symbol list may be incorrect
```

Currently, I get the following errors when attempting to use publicablly exported classes in `langchain_community.emeddings`:

```python
import langchain_community.embeddings

langchain_community.embeddings.HuggingFaceEmbeddings(...)  #  error: "HuggingFaceEmbeddings" is not exported from module "langchain_community.embeddings" (reportPrivateImportUsage)
```

This is solved easily by removing the dynamic expression.

### CODE REVIEW

**Comment by ringohoffman** [comment]
@baskaryan @eyurtsev @hwchase17

**Comment by baskaryan** [comment]
out of curiosity does `from langchain_community.embeddings import HuggingFaceEmbeddings` also cause issues?

**Comment by ringohoffman** [comment]
> out of curiosity does `from langchain_community.embeddings import HuggingFaceEmbeddings` also cause issues?

Yes


Screenshot 2024-05-20 at 17 16 09

**Review by eyurtsev — APPROVED** [review]

**Comment by ringohoffman** [comment]
@eyurtsev I tried fixing the linting errors that arose in `libs/community/tests/integration_tests/embeddings/test_titan_takeoff.py`. Should we retry?

**Comment by ringohoffman** [comment]
Okay I ran mypy locally to be sure it passes now:

```console
$ mypy libs/community/tests/integration_tests/embeddings/test_titan_takeoff.py
Success: no issues found in 1 source file
```

File: libs/community/langchain_community/embeddings/titan_takeoff.py [hunk 1]
```diff
@@ -178,12 +179,12 @@ def _embed(
         return self.client.embed(input, consumer_group)
 
     def embed_documents(
-        self, texts: List[str], consumer_group: Optional[str] = None
+        self, texts: Union[List[str], str], consumer_group: Optional[str] = None
```
Comment (baskaryan) [inline libs/community/langchain_community/embeddings/titan_takeoff.py:182]: this should only accept List[str]

Comment (ringohoffman) [reply libs/community/langchain_community/embeddings/titan_takeoff.py:182]: The tests supply str and they are passing:
https://github.com/langchain-ai/langchain/blob/eb096675a84fa08d7d75d532e765d19127c18b1a/libs/community/tests/integration_tests/embeddings/test_titan_takeoff.py#L27

https://github.com/langchain-ai/langchain/blob/eb096675a84fa08d7d75d532e765d19127c18b1a/libs/community/tests/integration_tests/embeddings/test_titan_takeoff.py#L61

`embed_tokens` calls `_embed`:

https://github.com/langchain-ai/langchain/blob/eb096675a84fa08d7d75d532e765d19127c18b1a/libs/community/langchain_community/embeddings/titan_takeoff.py#L180-L193

`_embed` accepts `Union[List[str], str]`:
https://github.com/langchain-ai/langchain/blob/eb096675a84fa08d7d75d532e765d19127c18b1a/libs/community/langchain_community/embeddings/titan_takeoff.py#L143-L145

So I do believe this is correct.

Comment (baskaryan) [reply libs/community/langchain_community/embeddings/titan_takeoff.py:182]: it should follow the base Embeddings abtraction https://github.com/langchain-ai/langchain/blob/eb096675a84fa08d7d75d532e765d19127c18b1a/libs/core/langchain_core/embeddings/embeddings.py#L12

Comment (ringohoffman) [reply libs/community/langchain_community/embeddings/titan_takeoff.py]: Okay, I've reverted this and updated the tests

**Comment by ringohoffman** [comment]
Are we good to merge? Thanks for the reviews @baskaryan and @eyurtsev!

**Pull request merged**